test/distributed/test_c10d_gloo.py

```diff
@@ -0,0 +1,2130 @@
+            else:
+                work.append(pg.gather([], input, opts))
+
+        # Wait for work to complete
+        expected = [torch.tensor([rank]) for rank in range(self.world_size)]
+        for i in range(self.world_size):
+            work[i].wait()
+            if i == self.rank:
+                self.assertEqual(expected, outputs)
+
+    def test_gather_basics(self):
+        self._test_gather_basics(lambda t: t.clone())
+
+    @skip_if_lt_x_gpu(2)
+    def test_gather_basics_cuda(self):
+        self._test_gather_basics(lambda t: t.clone().cuda())
+
+    def _test_gather_stress(self, inputs, fn):
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = c10d.ProcessGroupGloo(
+            store, self.rank, self.world_size, self.opts(threads=8)
+        )
+        work_handles = []
+        outputs = [
+            [[fn(torch.tensor([-1])) for _ in range(self.world_size)]]
+            for _ in range(len(inputs))
+        ]
+        expected_outputs = [
+            [[torch.tensor([i + j]) for j in range(self.world_size)]]
+            for i in range(len(inputs))
+        ]
+        for i in range(len(inputs)):
+            for root in range(self.world_size):
+                opts = c10d.GatherOptions()
+                opts.rootRank = root
+                if root == self.rank:
+                    work = pg.gather(outputs[i], [fn(inputs[i])], opts)
+                else:
+                    work = pg.gather([], [fn(inputs[i])], opts)
+                work_handles.append(work)
+
+        for i, work_handle in enumerate(work_handles):
+            work_handle.wait()
+            iter = i // self.world_size
+            root = i % self.world_size
+            if root == self.rank:
+                self.assertEqual(
+                    expected_outputs[iter],
+                    outputs[iter],
+                    msg=("Mismatch in iteration %d for root %d" % (iter, root)),
+                )
+
+    def test_gather_stress(self):
+        inputs = [torch.tensor([i + self.rank]) for i in range(1000)]
+        self._test_gather_stress(inputs, lambda t: t.clone())
+
+    @skip_if_lt_x_gpu(2)
+    def test_gather_stress_cuda(self):
+        inputs = [torch.tensor([i + self.rank]).cuda() for i in range(1000)]
+        self._test_gather_stress(inputs, lambda t: t.clone().cuda())
+
+    def test_allgather_checks(self):
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = c10d.ProcessGroupGloo(store, self.rank, self.world_size, self.opts())
+
+        t1 = torch.zeros([1], dtype=torch.float32)
+        t2 = torch.zeros([1], dtype=torch.float64)
+        t3 = torch.zeros([2], dtype=torch.float32)
+
+        with self.assertRaisesRegex(ValueError, "requires non-empty input tensor list"):
+            pg.allgather([], [])
+
+        with self.assertRaisesRegex(
+                ValueError, "requires input/output tensor lists to have the same length"
```

Comment:
The indentation here and many "ValueError" lines are changed. Previously it's aligned with the end of "with". Is this intentional?